src/prefect/core/task.py

```diff
@@ -953,7 +953,7 @@ def __getitem__(self, key: Any) -> "Task":
             - Task
         """
         return prefect.tasks.core.operators.GetItem(
-            checkpoint=self.checkpoint, result=self.result
+            checkpoint=self.checkpoint, name=f"{self.name}[{key}]", result=self.result
```

Comment:
Suggestion to use `repr` instead of `str` for the key, so that string-based indices still display with their quotes.

```suggestion
            checkpoint=self.checkpoint, name=f"{self.name}[{key!r}]", result=self.result
```